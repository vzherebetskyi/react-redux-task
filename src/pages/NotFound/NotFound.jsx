import React from 'react';
import { useNavigate } from 'react-router-dom';

import Layout from '../../components/layouts/base';
import './NotFound.scss';

const NotFound = () => {
  const navigate = useNavigate();

  return (
    <Layout>
      <div className="page-wrapper">
        <p className="page-title">Page not found</p>
        <button className="back-btn" onClick={() => navigate('/')}>
          Go back to the articles list
        </button>
      </div>
    </Layout>
  );
};

export default NotFound;
