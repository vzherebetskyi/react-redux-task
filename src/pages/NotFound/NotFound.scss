.page-wrapper {
  height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 100rem;
  margin: 0 auto;

  @media screen and (max-width: 1000px) {
    padding: 0 10px;
  }
}

.page-title {
  font-size: 1.2rem;
  font-weight: bold;
}

.back-btn {
  padding: 4px 8px;
  background: none;
  border: 1px solid #000;

  &:hover {
    cursor: pointer;
  }
}
