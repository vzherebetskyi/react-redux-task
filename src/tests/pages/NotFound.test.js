import React from 'react';
import renderer from 'react-test-renderer';

import NotFound from '../../pages/NotFound/NotFound';

const mockUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUsedNavigate,
}));

test('should render NotFound component correctly', () => {
  const elem = renderer.create(<NotFound />).toJSON();
  expect(elem).toMatchSnapshot();
});
