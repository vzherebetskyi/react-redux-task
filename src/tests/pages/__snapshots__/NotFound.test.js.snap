// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotFound component correctly 1`] = `
<div>
  <div
    className="page-wrapper"
  >
    <p
      className="page-title"
    >
      Page not found
    </p>
    <button
      className="back-btn"
      onClick={[Function]}
    >
      Go back to the articles list
    </button>
  </div>
  <div
    className="Toastify"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</div>
`;
